### PR TITLE
powershell: 7.1.4 -> 7.2.0

### PR DESCRIPTION
**Pull request by starcraft66**
###### Motivation for this change

This release adds support for .NET Core 6 and Apple Silicon (aarch64-darwin)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Seems like linux has a new runtime dependency on `systemtap` now:

Build fails with:

```
autoPatchelfHook could not satisfy dependency liblttng-ust.so.0 wanted by /nix/store/x5znpnviwinda1sai7jp4c9x0a7srfax-powershell-7.2.0/share/powershell/libcoreclrtraceptprovider.so
Add the missing dependencies to the build inputs or set autoPatchelfIgnoreMissingDeps=true
````

**Comment by starcraft66** [comment]
Okay, I will further test this on my Linux box at home, I just quickly worked on this on my macbook at work 😅

**Comment by Mic92** [comment]
It's actually `lttng-ust` that needs to go into `buildInputs`

File: pkgs/shells/powershell/default.nix [hunk 1]
```diff
@@ -49,10 +50,6 @@ stdenv.mkDerivation rec {
   '' + lib.optionalString (!stdenv.isDarwin && !stdenv.isAarch64) ''
     patchelf --replace-needed libcrypto${ext}.1.0.0 libcrypto${ext}.1.1 $pslibs/libmi.so
     patchelf --replace-needed libssl${ext}.1.0.0 libssl${ext}.1.1 $pslibs/libmi.so
-  '' + lib.optionalString (!stdenv.isDarwin) ''
-    # Remove liblttng-ust from dependencies once
-    # https://github.com/PowerShell/PowerShell/pull/14688 is in a release
-    patchelf --replace-needed liblttng-ust${ext}.0 liblttng-ust${ext}.1 $pslibs/libcoreclrtraceptprovider.so
```
Comment (Mic92) [inline pkgs/shells/powershell/default.nix:55]: Why did you remove this one?

Comment (starcraft66) [reply pkgs/shells/powershell/default.nix:55]: Yeah, I was misled by the comment. I will fix this up.

**Comment by starcraft66** [comment]
Okay, everything should be fine now, I tested compilation on aarch64-darwin and x86_64-linux.
